operator/pkg/apis/cassandra/v1beta1/seed_providers.go

```diff
@@ -0,0 +1,147 @@
+// Copyright DataStax, Inc.
+// Please see the included license file for details.
+
+package v1beta1
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/datastax/cass-operator/operator/pkg/utils"
+)
+
+type SeedProvider struct {
+	ClassName   string
+	Seeds       []string
+	OtherParams []map[string]interface{}
+}
+
+func buildClassNameForSeedProvider(serverType string) string {
+	var className string
+	if serverType == "cassandra" {
+		className = "org.apache.cassandra.locator.K8SeedProvider"
+	} else {
+		className = "org.apache.cassandra.locator.SimpleSeedProvider"
+	}
+	return className
+}
+
+func (dc *CassandraDatacenter) buildDefaultSeedProvider() *SeedProvider {
+	return &SeedProvider{
+		ClassName:   buildClassNameForSeedProvider(dc.Spec.ServerType),
+		Seeds:       dc.Spec.StaticSeedIPs,
+		OtherParams: nil,
+	}
+}
+
+func findSeedsAndOtherParams(parameters interface{}) ([]map[string]interface{}, []string) {
+	var otherParams []map[string]interface{}
+	var seeds []string
+
+	switch parameters := parameters.(type) {
+	case []map[string]interface{}:
+		for _, m := range parameters {
+			if v, ok := m["seeds"]; ok {
+				switch v := v.(type) {
+				case []string:
+					seeds = v
+				}
+			} else {
+				otherParams = append(otherParams, m)
```

Comment:
keep track of any other param maps other than seeds so we can preserve them when we do an update